src/gen_rules.ml

```diff
@@ -969,6 +969,41 @@ module Gen(P : Params) = struct
         ~preprocess:lib.preprocess
         ~virtual_deps:lib.virtual_deps
     in
+
+    (* Add a rule to build .merlin files *)
+    (match Path.extract_build_context dir with
+      | Some ("default", remaindir) ->
+        add_rule (
+          requires
+          >>>
+          Build.arr (fun libs ->
+            let source_dirs = ["S " ^ Path.reach dir ~from:dir] in
+            let build_dirs =
+              libs
+              |> Lib.include_paths
+              |> Path.Set.elements
+              |> List.map ~f:(fun path -> "B " ^ Path.reach path ~from:remaindir)
+            in
+            let build_dirs = ("B " ^ Path.reach dir ~from:remaindir) :: build_dirs in
+            let pkgs = List.map libs ~f:(fun lib -> "PKG " ^ Lib.best_name lib) in
```

Comment:
Ah, ok.  So, to address this, I found that the only way I could differentiate internal vs external libraries was to call into `Lib_db.interpret_lib_deps`, which was impossible because `t` is hidden outside the module, so I had to add a function inside the module to get access to that information.